cpp/src/parquet/column_reader.h

```diff
@@ -115,11 +116,30 @@ class PARQUET_EXPORT PageReader {
                                           bool always_compressed = false,
                                           const CryptoContext* ctx = NULLPTR);
 
+  // If skip_page_callback_ is present (not null), NextPage() will call the
+  // callback function exactly once per page in the order the pages appear in
+  // the column. If the callback function returns true the page will be
+  // skipped. The callback will be called only if the page type is DATA_PAGE or
+  // DATA_PAGE_V2. Dictionary pages will not be skipped.
+  // This setter must be called at most once to set the callback.
+  // \note API EXPERIMENTAL
+  void set_skip_page_callback(
+      std::function<bool(const DataPageStats&)> skip_page_callback) {
+    if (skip_page_callback_) {
+      throw ParquetException("set_skip_page_callback was called more than once");
```

Comment:
Why would this be forbidden?

Reply:
I was thinking mostly about why would we allow this? By forbidding it, we have one less scenario to think of in the code.

Reply:
Well, unless we do a similar thing for other configuration properties, this seems a bit gratuitous.

Reply:
Done.